File: hostloc2.php
<?php
$useList = array(
                                '账号1'=>'密码1',
                                '账号2'=>'密码2',
                                '账号3'=>'密码3',
                                );
header("Content-Type: text/html; charset=utf-8");
foreach($useList as $k=>$v){
        locSign($k,$v);
}
function locSign($use,$pwd){
        $html = http_post('https://www.hostloc.com/member.php?mod=logging&action=login&loginsubmit=yes&infloat=yes&lssubmit=yes&inajax=1','fastloginfield=username&username='.urlencode($use).'&cookietime=2592000&password='.urlencode($pwd).'&quickforward=yes&handlekey=ls');
        if(!strrpos($html,'window.location.href')){echo $use.' login error<br>';return false;}
        $cookie = getCookie($html,'Set-Cookie: ',';');
        for($i=26200; $i<=26210; $i++){
                $html = http_get("https://www.hostloc.com/space-uid-{$i}.html",$cookie);
        }
        echo $use.' Sign ok!!!<br>';
}
function http_get($url,$cookie) {
        $ch = curl_init($url);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
        curl_setopt($ch, CURLOPT_HTTPHEADER,array('User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0','Referer: https://www.hostloc.com/forum-45-1.html','Cookie: '.$cookie));
        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
        $src = curl_exec($ch);
        curl_close($ch);
        return $src;
}
function http_post($url,$body){
        $ch = curl_init();  
        curl_setopt($ch, CURLOPT_URL,$url);
        curl_setopt($ch, CURLOPT_HEADER, 1);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);  
        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);  
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); 
        curl_setopt($ch, CURLOPT_HTTPHEADER,array('User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0','Referer: https://www.hostloc.com/forum.php'));
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $body);        
        $result = curl_exec($ch);  
        curl_close($ch);  
        return $result;    
}
function getCookie($str,$leftStr,$rightStr){
        if(strrpos($str,$leftStr) == false || strrpos($str,$rightStr) == false){return '';}
        $arr1 = explode($leftStr,$str);$i = 0;$cookie = '';
        foreach($arr1 as $k=>$v){$i++;if($i > 1){$arr2 = explode($rightStr,$v);$cookie .= $arr2[0].'; ';}}
        return $cookie;
}

File: hostloc2.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

import os
from urllib import request
from http import cookiejar

hostloc_username = os.getenv('hostloc_username')
hostloc_password = os.getenv('hostloc_password')

def Login(URL, UserData):
    __cookies = ''
    __cookie = cookiejar.CookieJar()
    __handler = request.HTTPCookieProcessor(__cookie)
    __req = request.Request(URL, data=str(UserData).encode('utf-8'))
    request.build_opener(__handler).open(__req)
    for cookie in __cookie:
        __cookies += cookie.name + '=' + cookie.value + ';'
    return __cookies


def GetPage(URL, Header_Cookies):
    __Header = {'Cookie': str(Header_Cookies)}
    __req = request.Request(URL, headers=__Header)
    return request.urlopen(__req).read().decode('utf-8')


def GetCredit(username, password):
    Login_URL = 'https://www.hostloc.com/member.php?mod=logging&action=login&loginsubmit=yes&infloat=yes&lssubmit=yes&inajax=1'
    My_Home = 'https://www.hostloc.com/home.php?mod=spacecp&inajax=1'

    user_data = 'username=' + str(username) + '&' + 'password=' + str(password)
    My_Cookies = Login(Login_URL, user_data)

    if '<td>' + str(username) + '</td>' not in GetPage(My_Home, My_Cookies):
        isLogin = False
        print('[%s] Login Fail!' % username)
    else:
        isLogin = True
        print('[%s] Login Success!' % username)

    if isLogin:
        for __x in range(25297, 25309):
            __url = 'https://www.hostloc.com/space-uid-{}.html'.format(__x)
            GetPage(__url, My_Cookies)

def start():
    GetCredit(hostloc_username, hostloc_password)

if __name__ == '__main__':
    GetCredit(hostloc_username, hostloc_password)
